src/OpenLoco/Windows/PromptBrowseWindow.cpp

```diff
@@ -402,11 +410,39 @@ namespace OpenLoco::Ui::Windows::PromptBrowse
         self->widgets[widx::parent_button].left = self->width - 26;
         self->widgets[widx::parent_button].right = self->width - 3;
 
-        // Resume the original prepare_draw routine beyond the widget repositioning.
-        registers regs;
-        regs.edi = X86Pointer(buffer);
-        regs.esi = X86Pointer(self);
-        call(0x00445D91, regs);
+        // Get width of the base 'Folder:' string
+        auto args = FormatArguments::common(StringIds::empty);
+        char folderBuffer[256]{};
+        StringManager::formatString(folderBuffer, StringIds::window_browse_folder, &args);
+        Gfx::setCurrentFontSpriteBase(224);
+        const auto folderLabelWidth = Gfx::getStringWidth(folderBuffer);
+
+        // We'll ensure the folder width does not reach the parent button.
+        const uint16_t maxWidth = self->widgets[widx::parent_button].left - folderLabelWidth - 10;
+        strncpy(&_displayFolderBuffer[0], &_directory[0], 512);
+        uint16_t folderWidth = Gfx::getStringWidth(_displayFolderBuffer);
+
+        // If the folder already fits, we're done.
+        if (folderWidth <= maxWidth)
+            return;
+
+        char* relativeDirectory = &_directory[0];
```

Comment:
`fs::path` has begin and end iterators that can be used to split by folder.
An alternative algorithm would be to iterate on folder, get the stringWidth of each folder + seperator, then reverse the result to work out how many parts are required. Finally can then build up the directory.